providers/implementations/signature/ecdsa_sig.c.in

```diff
@@ -672,153 +675,144 @@ static void *ecdsa_dupctx(void *vctx)
 
-    p = OSSL_PARAM_locate(params, OSSL_SIGNATURE_PARAM_ALGORITHM_ID);
-    if (p != NULL && !OSSL_PARAM_set_octet_string(p,
-                                                  ctx->aid_len == 0 ? NULL : ctx->aid_buf,
-                                                  ctx->aid_len))
+    if (p.algid != NULL
+            && !OSSL_PARAM_set_octet_string(p.algid,
+                                            ctx->aid_len == 0 ? NULL : ctx->aid_buf,
+                                            ctx->aid_len))
         return 0;
 
-    p = OSSL_PARAM_locate(params, OSSL_SIGNATURE_PARAM_DIGEST_SIZE);
-    if (p != NULL && !OSSL_PARAM_set_size_t(p, ctx->mdsize))
+    if (p.size != NULL && !OSSL_PARAM_set_size_t(p.size, ctx->mdsize))
         return 0;
 
-    p = OSSL_PARAM_locate(params, OSSL_SIGNATURE_PARAM_DIGEST);
-    if (p != NULL && !OSSL_PARAM_set_utf8_string(p, ctx->md == NULL
-                                                    ? ctx->mdname
-                                                    : EVP_MD_get0_name(ctx->md)))
+    if (p.digest != NULL
+            && !OSSL_PARAM_set_utf8_string(p.digest, ctx->md == NULL
+                                                     ? ctx->mdname
+                                                     : EVP_MD_get0_name(ctx->md)))
         return 0;
 
-    p = OSSL_PARAM_locate(params, OSSL_SIGNATURE_PARAM_NONCE_TYPE);
-    if (p != NULL && !OSSL_PARAM_set_uint(p, ctx->nonce_type))
+    if (p.nonce != NULL && !OSSL_PARAM_set_uint(p.nonce, ctx->nonce_type))
         return 0;
 
 #ifdef FIPS_MODULE
-    p = OSSL_PARAM_locate(params, OSSL_SIGNATURE_PARAM_FIPS_VERIFY_MESSAGE);
-    if (p != NULL && !OSSL_PARAM_set_uint(p, ctx->verify_message))
+    if (p.verify != NULL && !OSSL_PARAM_set_uint(p.verify, ctx->verify_message))
         return 0;
 #endif
 
-    if (!OSSL_FIPS_IND_GET_CTX_PARAM(ctx, params))
+    if (!OSSL_FIPS_IND_GET_CTX_FROM_PARAM(ctx, p.ind))
         return 0;
     return 1;
 }
 
-static const OSSL_PARAM known_gettable_ctx_params[] = {
-    OSSL_PARAM_octet_string(OSSL_SIGNATURE_PARAM_ALGORITHM_ID, NULL, 0),
-    OSSL_PARAM_size_t(OSSL_SIGNATURE_PARAM_DIGEST_SIZE, NULL),
-    OSSL_PARAM_utf8_string(OSSL_SIGNATURE_PARAM_DIGEST, NULL, 0),
-    OSSL_PARAM_uint(OSSL_SIGNATURE_PARAM_NONCE_TYPE, NULL),
-#ifdef FIPS_MODULE
-    OSSL_PARAM_uint(OSSL_SIGNATURE_PARAM_FIPS_VERIFY_MESSAGE, NULL),
-#endif
-    OSSL_FIPS_IND_GETTABLE_CTX_PARAM()
-    OSSL_PARAM_END
-};
-
 static const OSSL_PARAM *ecdsa_gettable_ctx_params(ossl_unused void *vctx,
                                                    ossl_unused void *provctx)
 {
-    return known_gettable_ctx_params;
+    return ecdsa_get_ctx_params_list;
 }
 
+struct ecdsa_all_set_ctx_params_st {
+    OSSL_PARAM *digest;     /* ecdsa_set_ctx_params */
+    OSSL_PARAM *propq;      /* ecdsa_set_ctx_params */
+    OSSL_PARAM *size;       /* ecdsa_set_ctx_params */
+    OSSL_PARAM *ind_d;
+    OSSL_PARAM *ind_k;
+    OSSL_PARAM *kat;
+    OSSL_PARAM *nonce;
```

Comment:
Not overlyjoyed by these abbreviated names
e.g. nonce != nonce_type (Others dont care so I will just comment that I dont like it :))